compiler-plugin/src/main/java/io/ballerina/stdlib/persist/compiler/PersistRecordValidator.java

```diff
@@ -770,13 +796,17 @@ private void validFieldTypeAndRelation(RecordTypeDescriptorNode recordNode,
             String tableAssociationType = "";
             boolean isArrayType = false;
             boolean isUserDefinedType = false;
+            boolean isOptionalField = false;
```

Comment:
I will be removing validFieldTypeAndRelation function as part of relation check reformatting, I have seperated out field type check in `validateEntityFields`. Lets fix it there